bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -274,7 +274,9 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
         switch (targetTypeTag) {
             case TypeTags.UNION_TAG:
                 for (Type memType : ((BUnionType) targetType).getMemberTypes()) {
-                    if (TypeChecker.getType(inputValue) == memType) {
+                    if (TypeChecker.getType(inputValue) == memType
+                            || isIntegerSubtypeAndConvertible(inputValue, memType)
+                    ) {
```

Comment:
Should this closing parenthesis need goto line 278. seems format error

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/34136/commits/0af5ad271764dfb727df0e90bda5905cbb4ed362